src/mon/Monitor.cc

```diff
@@ -4844,16 +4846,19 @@ bool Monitor::_scrub(ScrubResult *r,
     }
 
     bufferlist bl;
-    //TODO: what when store->get returns error or empty bl?
-    store->get(k.first, k.second, bl);
+    int err = store->get(k.first, k.second, bl);
+    assert(err == 0 && bl.length() > 0);
```

Comment:
@tchaikov you're right. I think `bl` may be empty during a scrub (not so much during a sync though, as above), but we should take into account keys that may have empty values. My oversight on this one could have lead to an annoying regression. *sigh*

Thanks for catching it.